Uncaught exceptions in PQBinReader, FlowSightReader

Hi all,

Here are two uncaught exceptions I see frequently when loading images:

```
Exception in thread "Thread-1" java.io.EOFException: Attempting to read beyond end of file.
	at loci.common.NIOFileHandle.readFloat(NIOFileHandle.java:352)
	at loci.common.RandomAccessInputStream.readFloat(RandomAccessInputStream.java:501)
	at loci.formats.in.PQBinReader.isThisType(PQBinReader.java:103)
Caused by: java.nio.BufferUnderflowException
	at java.nio.Buffer.nextGetIndex(Buffer.java:506)
	at java.nio.HeapByteBuffer.getFloat(HeapByteBuffer.java:463)
	at loci.common.NIOFileHandle.readFloat(NIOFileHandle.java:350)
	... 2 more
Caught exception in class loci.formats.in.PQBinReader.isThisType
```

```
Exception in thread "Thread-1" java.lang.IllegalArgumentException
	at java.nio.Buffer.position(Buffer.java:244)
	at loci.common.NIOFileHandle.buffer(NIOFileHandle.java:605)
	at loci.common.NIOFileHandle.seek(NIOFileHandle.java:293)
	at loci.common.RandomAccessInputStream.seek(RandomAccessInputStream.java:174)
	at loci.formats.tiff.TiffParser.<init>(TiffParser.java:110)
	at loci.formats.in.FlowSightReader.isThisType(FlowSightReader.java:99)
Caught exception in class loci.formats.in.FlowSightReader.isThisType
```

I am able to load images of any supported type. Allowing these exceptions to raise causes a lot of unnecessary noise.

I plan to look into this further in the next few days. However, my Java is quite rusty and I am unfamiliar with the inner workings of this project. If any more experienced contributors have suggestions, I'd love to hear them. :)

Best,
Claire

--- a/components/formats-bsd/src/loci/formats/in/FlowSightReader.java
+++ b/components/formats-bsd/src/loci/formats/in/FlowSightReader.java
@@ -95,8 +95,13 @@ public class FlowSightReader extends FormatReader {
    */
   @Override
   public boolean isThisType(RandomAccessInputStream stream)
-      throws IOException {
-    TiffParser tiffParser = new TiffParser(stream);
+          throws IOException {
+    TiffParser tiffParser;
+    try {
+      tiffParser = new TiffParser(stream);
+    } catch (java.lang.IllegalArgumentException iae) {
+      return false;
+    }
     if (! tiffParser.isValidHeader()) return false;
     IFD ifd = tiffParser.getFirstIFD();
     if (ifd == null) return false;

--- a/components/formats-gpl/src/loci/formats/in/PQBinReader.java
+++ b/components/formats-gpl/src/loci/formats/in/PQBinReader.java
@@ -93,21 +93,25 @@ public class PQBinReader extends FormatReader {
   /* @see loci.formats.IFormatReader#isThisType(RandomAccessInputStream) */
   @Override
   public boolean isThisType(RandomAccessInputStream stream) throws IOException {
-    
-    long fileLength = stream.length();
     int bpp = FormatTools.getBytesPerPixel(FormatTools.UINT32);
     stream.order(true);
+
     // Header
-    int sizeX = stream.readInt();
-    int sizeY = stream.readInt();
-    float pixResol = stream.readFloat();  // resolution of time axis of every Decay (in ns)
-    int sizeT = stream.readInt();     
-    if (( sizeX * sizeY * sizeT * bpp) + HEADER_SIZE  == fileLength)  {
-      return true;
-    }
-    else  {
+    long fileLength;
+    int sizeX;
+    int sizeY;
+    int sizeT;
+    try {
+      fileLength = stream.length();
+      sizeX = stream.readInt();
+      sizeY = stream.readInt();
+      stream.readFloat();  // resolution of time axis of every Decay (in ns)
+      sizeT = stream.readInt();
+    } catch (java.io.EOFException eof) {
       return false;
-    } 
+    }
+
+    return (sizeX * sizeY * sizeT * bpp) + HEADER_SIZE == fileLength;
   }
 
   /**

